### PR TITLE
use deleteRecursively

### PR DESCRIPTION
**Pull request by elharo**

### CODE REVIEW

**Comment by chanseokoh** [comment]
Worth looking at why the tests work differently.

File: plugins/com.google.cloud.tools.eclipse.util.test/src/com/google/cloud/tools/eclipse/util/status/StatusUtilTest.java [hunk 1]
```diff
@@ -99,14 +98,14 @@ public void testError_withClassAndException() {
 
   @Test
   public void testError_withInstance() {
-    IStatus error = StatusUtil.error(new DeleteAllVisitor(), "test error msg");
+    IStatus error = StatusUtil.error("an object", "test error msg");
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.util.test/src/com/google/cloud/tools/eclipse/util/status/StatusUtilTest.java]: I think you can just use `this`: the test class is hosted by bundle.

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/CleanupOldDeploysJob.java [hunk 2]
```diff
@@ -65,8 +64,8 @@ protected IStatus run(IProgressMonitor monitor) {
   }
 
   private void deleteDirectories(List<File> directories) throws IOException {
-    for (int i = RECENT_DIRECTORIES_TO_KEEP; i < directories.size(); ++i) {
-      Files.walkFileTree(directories.get(i).toPath(), new DeleteAllVisitor());
+    for (int i = RECENT_DIRECTORIES_TO_KEEP; i < directories.size(); i++) {
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/CleanupOldDeploysJob.java]: I guess that `<` should be `<=`.

Comment (chanseokoh) [reply plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/CleanupOldDeploysJob.java]: I'm wondering why the behavior changed.

**Comment by elharo** [comment]
The problem was I was calling the wrong method to delete the contents but not the directory.

File: plugins/com.google.cloud.tools.eclipse.util.test/src/com/google/cloud/tools/eclipse/util/status/StatusUtilTest.java (hunk 1 above)
Comment (chanseokoh) [reply plugins/com.google.cloud.tools.eclipse.util.test/src/com/google/cloud/tools/eclipse/util/status/StatusUtilTest.java]: This one and the below are still `"an object"`.

**Comment by elharo** [comment]
```
37217 [ERROR] Failed to resolve target definition /home/travis/build/GoogleCloudPlatform/google-cloud-eclipse/eclipse/neon/gcp-eclipse-neon.target: Could not find "org.slf4j.api/1.7.10.v20170224-1220" in the repositories of the current location -> [Help 1]
```

Is this because of the now fixed target platform issues? Can I rebase to fix?

**Comment by briandealwis** [comment]
Merging master should have fixed it.

**Comment by elharo** [comment]
Failure on Kokoro does look related:

```
Running com.google.cloud.tools.eclipse.appengine.deploy.CleanupOldDeploysJobTest
Tests run: 4, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.031 sec <<< FAILURE! - in com.google.cloud.tools.eclipse.appengine.deploy.CleanupOldDeploysJobTest
testRun_withThreeDirectories(com.google.cloud.tools.eclipse.appengine.deploy.CleanupOldDeploysJobTest)  Time elapsed: 0 sec  <<< FAILURE!
java.lang.AssertionError: 
Expected: is <2>
     but: was <3>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at com.google.cloud.tools.eclipse.appengine.deploy.CleanupOldDeploysJobTest.testRun(CleanupOldDeploysJobTest.java:67)
	at com.google.cloud.tools.eclipse.appengine.deploy.CleanupOldDeploysJobTest.testRun_withThreeDirectories(CleanupOldDeploysJobTest.java:56)
```

**Comment by elharo** [comment]
From the Guava release notes:

This includes deleteRecursively and deleteDirectoryContents methods which are secure against race conditions that Java previously had no way of dealing with provided that the FileSystem supports SecureDirectoryStream (modern Linux versions do; Windows [NTFS at least] does not). For security, these will throw an exception if SecureDirectoryStream is not supported unless RecursiveDeleteOption.ALLOW_INSECURE is passed when calling the method.

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/CleanupOldDeploysJob.java (hunk 2 above)
Comment (briandealwis) [reply plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/CleanupOldDeploysJob.java]: `++i` vs `i++`

Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/CleanupOldDeploysJob.java]: https://www.ibm.com/developerworks/library/j-noaccent/

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.test/src/com/google/cloud/tools/eclipse/appengine/deploy/CleanupOldDeploysJobTest.java [hunk 3]
```diff
@@ -62,17 +63,17 @@ private void testRun(int directoryCount, String[] expectedDirectoriesToKeep) thr
 
     IPath tempFolderPath = new Path(tempFolder.getRoot().toString());
     CleanupOldDeploysJob job = new CleanupOldDeploysJob(tempFolderPath);
-    job.run(mock(IProgressMonitor.class));
+    assertEquals(Status.OK_STATUS, job.run(new NullProgressMonitor()));
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.test/src/com/google/cloud/tools/eclipse/appengine/deploy/CleanupOldDeploysJobTest.java]: You should be able to use `null` for the progress monitor (though this is perfectly acceptable)

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/CleanupOldDeploysJob.java (hunk 2 above)
Comment (briandealwis) [reply plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/CleanupOldDeploysJob.java]: Oh oops!  I was wrongly thinking the pre-post increment happened during the evaluation of the comparison expression.  Please ignore (:blush:)

**Pull request merged**